lxd/storage/backend_lxd.go

```diff
@@ -7778,3 +7869,26 @@ func (b *lxdBackend) CreateCustomVolumeFromBackup(srcBackup backup.Info, srcData
 	revert.Success()
 	return nil
 }
+
+// getSnapshotParentUUID returns the UUID of the snapshot's parent volume.
+func (b *lxdBackend) getSnapshotParentUUID(vol drivers.Volume, projectName string) (string, error) {
+	parentName, _, isSnapshot := api.GetParentAndSnapshotName(vol.Name())
+	if !isSnapshot {
+		// Volume has no parent.
+		return "", nil
```

Comment:
should this be an error as it means the vol passed in is not a snapshot and the function is being used incorrectly?

Reply:
I was thinking about this one, and decided to just return an empty string to avoid checking for snapshot on every place where the function is called. However, in such case the function name "GetParentVolumeUUID" would probably be more appropirate, as non-snapshot volumes have no parent.

On the other side, returning an error as you have suggested prevents unnecessary or incorrect usage of the function.

Reply:
> Rename to GetParentVolumeUUID